File: examples/nodejs/send_sms.js
/**
 * Selfhost SMS Gateway - Node.js Example
 * https://selfhost.smpp.asia
 */

class SelfhostSMS {
  constructor(token) {
    this.baseUrl = 'https://selfhost.smpp.asia';
    this.token = token;
  }

  async sendSms(phone, message) {
    return this.request('POST', '/api/sms/send', { phone, message });
  }

  async requestOtp(phone, brand = 'App') {
    return this.request('POST', '/api/otp/request', { phone, brand });
  }

  async verifyOtp(phone, code) {
    return this.request('POST', '/api/otp/verify', { phone, code });
  }

  async getHistory(limit = 50) {
    return this.request('GET', `/api/sms/history?limit=${limit}`);
  }

  async getProfile() {
    return this.request('GET', '/api/client/profile');
  }

  async request(method, endpoint, data = null) {
    const options = {
      method,
      headers: {
        'Authorization': `Bearer ${this.token}`,
        'Content-Type': 'application/json'
      }
    };

    if (data && method === 'POST') {
      options.body = JSON.stringify(data);
    }

    const response = await fetch(`${this.baseUrl}${endpoint}`, options);
    return response.json();
  }
}

// ============================================
// USAGE EXAMPLE
// ============================================

const TOKEN = 'YOUR_TOKEN_HERE'; // Get from dashboard
const sms = new SelfhostSMS(TOKEN);

async function main() {
  // Send SMS
  let result = await sms.sendSms('09xxxxxxxxx', 'Hello from Node.js!');
  console.log('Send SMS:', result);

  // Request OTP
  result = await sms.requestOtp('09xxxxxxxxx', 'MyApp');
  console.log('Request OTP:', result);

  // Verify OTP
  result = await sms.verifyOtp('09xxxxxxxxx', '123456');
  console.log('Verify OTP:', result);

  // Get profile
  result = await sms.getProfile();
  console.log('Profile:', result);
}

main().catch(console.error);

module.exports = SelfhostSMS;

File: examples/python/send_sms.py
"""
Selfhost SMS Gateway - Python Example
https://selfhost.smpp.asia

Install: pip install requests
"""

import requests

class SelfhostSMS:
    def __init__(self, token):
        self.base_url = 'https://selfhost.smpp.asia'
        self.headers = {
            'Authorization': f'Bearer {token}',
            'Content-Type': 'application/json'
        }

    def send_sms(self, phone, message):
        """Send SMS"""
        response = requests.post(
            f'{self.base_url}/api/sms/send',
            headers=self.headers,
            json={'phone': phone, 'message': message}
        )
        return response.json()

    def request_otp(self, phone, brand='App'):
        """Request OTP"""
        response = requests.post(
            f'{self.base_url}/api/otp/request',
            headers=self.headers,
            json={'phone': phone, 'brand': brand}
        )
        return response.json()

    def verify_otp(self, phone, code):
        """Verify OTP"""
        response = requests.post(
            f'{self.base_url}/api/otp/verify',
            headers=self.headers,
            json={'phone': phone, 'code': code}
        )
        return response.json()

    def get_history(self, limit=50):
        """Get SMS history"""
        response = requests.get(
            f'{self.base_url}/api/sms/history?limit={limit}',
            headers=self.headers
        )
        return response.json()

    def get_profile(self):
        """Get account profile"""
        response = requests.get(
            f'{self.base_url}/api/client/profile',
            headers=self.headers
        )
        return response.json()


# ============================================
# USAGE EXAMPLE
# ============================================

if __name__ == '__main__':
    TOKEN = 'YOUR_TOKEN_HERE'  # Get from dashboard
    sms = SelfhostSMS(TOKEN)

    # Send SMS
    result = sms.send_sms('09xxxxxxxxx', 'Hello from Python!')
    print('Send SMS:', result)

    # Request OTP
    result = sms.request_otp('09xxxxxxxxx', 'MyApp')
    print('Request OTP:', result)

    # Verify OTP
    result = sms.verify_otp('09xxxxxxxxx', '123456')
    print('Verify OTP:', result)

    # Get profile
    result = sms.get_profile()
    print('Profile:', result)
